src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -62,9 +62,22 @@ private void AddArrayRank(IArrayTypeSymbol symbol)
 
             AddPunctuation(SyntaxKind.OpenBracketToken);
 
-            if (insertStars && symbol.Rank > 1)
+            if (symbol.Rank > 1)
             {
-                AddPunctuation(SyntaxKind.AsteriskToken);
+                if (insertStars)
+                {
+                    AddPunctuation(SyntaxKind.AsteriskToken);
+                }
+            }
+            else
+            {
+                var array = symbol as ArrayTypeSymbol;
```

Comment:
Yes, or some other implementation of IArrayTypeSymbol. Once we are ready to make public API changes, IsSZArray property must become part of IArrayTypeSymbol, then we should simplify this code to use it instead.